samples/audio/sof/README.md

```diff
@@ -0,0 +1,305 @@
+source directory, so it can be accessed in the docker environment.
+
+```
+$ cd ~/work/sof/sof
+$ mkdir zephyr
+$ cp <zephyr dir>/build/zephyr/zephyr.elf zephyr/
+```
+
+Run SOF docker image and build the image
+
+```
+$ cd ~/work/sof/sof
+$ ./scripts/docker-run.sh bash
+#
+# Now shell in docker
+#
+$ xtensa-apl-elf-objcopy -O binary build_apl_gcc/src/platform/apollolake/base_module mod-apl.bin
+$ xtensa-apl-elf-objcopy --add-section .module=mod-apl.bin --set-section-flags .module=load,readonly zephyr/zephyr.elf
+$ build_apl_gcc/rimage_ep/build/rimage -o sof-apl.ri -p sof-apl.ldc -m apl -k rimage/keys/otc_private_key.pem -i 3 build_apl_gcc/src/arch/xtensa/bootloader-apl zephyr/zephyr.elf
+```
+
+# Using image on Qemu (SOF Qemu)
+
+## Install SOF Qemu
+
+This is Qemu with SOF patches for the APL xtensa. It is capable of running
+exactly the same code as a real hardware (ADSP on up_squared).
+
+### Get Qemu with docker
+
+```
+$ docker pull thesofproject/sofqemu
+$ docker tag thesofproject/sofqemu sofqemu
+```
+
+### Build Qemu from local sources
+
+Instructions are taken from docker build configuration in
+**scripts/docker_build/sof_qemu/Dockerfile**
+
+```
+$ git clone https://github.com/thesofproject/qemu.git
+$ cd qemu
+$ git checkout sof-stable
+$ ./configure --prefix=`pwd`/ --target-list=xtensa-softmmu --enable-coroutine-pool
+$ make
+```
+
+## Running and debugging with Qemu
+
+### Run image on Qemu
+
+```
+$ cd qemu
+$ ./xtensa-softmmu/qemu-system-xtensa -cpu broxton -M adsp_bxt -nographic -kernel ../sof.git/sof-apl.ri -rom ../sof.git/build_apl_gcc/src/arch/xtensa/rom-apl.bin -s -S -semihosting
+```
+
+Note that **-s** is a _shorthand for -gdb tcp::1234_ and **-S** is to _freeze
+CPU at startup and use ‘c’ to start execution_. Skip these options to run
+without debugger.
+
+Note that **-semihosting** enables logging to qemu console using simcall
+instructions with xtensa simulator backend.
+
+Use the **sof-apl.ri **created in the “_Create rimage_” section.
+
+Using helper script:
+
+```
+$ ./xtensa-host.sh apl -d -k ../sof/sof-apl.ri -r ../sof/build_apl_gcc/src/arch/xtensa/rom-apl.bin
+```
+
+Where **sof-apl.ri** is an image created with rimage tool combining bootloader
+and kernel, **rom-apl.bin** - is ROM image which should do the same boot
+sequence as xtensa DSP. Since we can run exactly the same images on Qemu and on
+a real **up_squared** hardware we can use this Qemu for early development.
+
+### Run image with Docker Qemu
+
+Use the following script to run the SOF Qemu docker image.
+
+```
+$ cd ~/work/sof/sof
+$ ./scripts/docker-qemu.sh bash
+```
+
+Then follow the commands above.
+
+### Qemu target - host communication
+
+Communication between Qemu running Zephyr and host Linux can be done through
+/dev/shm. For example if Zephyr code in the sample does something like
+
+```
+mailbox_sw_reg_write(SRAM_REG_ROM_STATUS, 0xabbac0fe);
+```
+
+In the Linux host we can verify value written with:
+
+```
```

Comment:
Id remove this one since the one with semihosting is quite a bit easier to read, besides I don't know if I had you backend pushed in this tree.